File: app/dbmodels/Contract.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Contract',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                     => DBModels::type('increments',
                                               array('visible' => false)),
    'is_deleted'             => DBModels::type('boolean',
                                               array('visible' => false)),
    'is_template'            => DBModels::type('boolean',
                                               array('visible' => false)),
    'template_name'          => DBModels::type('string',
                                               array('visible' => false)),
    'name'                   => DBModels::type('string'),
    'num'                    => DBModels::type('string'),
    'comment'                => DBModels::type('text'),
    'contract_type_id'        => DBModels::type('integer'),
    'begin_date'             => DBModels::type('date'),
    'duration'               => DBModels::type('integer'),
    'notice'                 => DBModels::type('integer'),
    'periodicity'            => DBModels::type('integer'),
    'billing'                => DBModels::type('integer'),
    'accounting_number'      => DBModels::type('string'),
    'week_begin_hour'        => DBModels::type('time'),
    'week_end_hour'          => DBModels::type('time'),
    'saturday_begin_hour'    => DBModels::type('time'),
    'saturday_end_hour'      => DBModels::type('time'),
    'use_saturday'           => DBModels::type('boolean'),
    'monday_begin_hour'      => DBModels::type('time'),
    'monday_end_hour'        => DBModels::type('time'),
    'use_monday'             => DBModels::type('boolean'),
    'max_links_allowed'      => DBModels::type('integer'),
    'alert'                  => DBModels::type('integer'),
    'renewal'                => DBModels::type('integer'),
);

$table['relationships'] = array(
    'contract_type' => array(
        'type' => 'belongsTo',
        'item' => 'ContractType'),
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/contract.py
from fusionglpi import db


class Contract(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    is_template = db.Column(db.Boolean, nullable=False, server_default="0")
    template_name = db.Column(db.String(255, None, True))
    name = db.Column(db.String(255, None, True))
    num = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    contract_type_id = db.Column(db.Integer,
                                 db.ForeignKey('contract_type.id'))
    begin_date = db.Column(db.Date)
    duration = db.Column(db.Integer, nullable=False, server_default="0")
    notice = db.Column(db.Integer, nullable=False, server_default="0")
    periodicity = db.Column(db.Integer, nullable=False, server_default="0")
    billing = db.Column(db.Integer, nullable=False, server_default="0")
    accounting_number = db.Column(db.String(255, None, True))
    week_begin_hour = db.Column(db.Time)
    week_end_hour = db.Column(db.Time)
    saturday_begin_hour = db.Column(db.Time)
    saturday_end_hour = db.Column(db.Time)
    use_saturday = db.Column(db.Boolean, nullable=False, server_default="0")
    monday_begin_hour = db.Column(db.Time)
    monday_end_hour = db.Column(db.Time)
    use_monday = db.Column(db.Boolean, nullable=False, server_default="0")
    max_links_allowed = db.Column(db.Integer, nullable=False,
                                  server_default="0")
    alert = db.Column(db.Integer, nullable=False, server_default="0")
    renewal = db.Column(db.Integer, nullable=False, server_default="0")
